File: gfless-nodejs/examples/basic_example.js
const { 
  createIdentity, 
  createBlackBox, 
  createEncryptedBlackBox,
  BlackBox,
  Utils,
  Crypto,
  getVersion 
} = require('../lib');

async function basicExample() {
  console.log('Gfless Node.js Library Basic Example');
  console.log('Version:', getVersion());
  console.log();

  try {
    // Create an identity
    const identity = createIdentity('identity.json');
    console.log('✓ Identity created');

    // Update the identity
    identity.update();
    console.log('✓ Identity updated');

    // Get fingerprint data
    const fingerprint = identity.getFingerprint();
    console.log('✓ Fingerprint retrieved');
    console.log('Fingerprint fields:', Object.keys(fingerprint).length);

    // Create a basic blackbox
    const blackBox = createBlackBox(identity, {
      app: 'example',
      version: '1.0'
    });
    
    const encoded = blackBox.encoded();
    console.log('✓ BlackBox encoded:', encoded.substring(0, 50) + '...');

    // Test static encode/decode
    const testData = '["test", 123, true]';
    const staticEncoded = BlackBox.encode(testData);
    const staticDecoded = BlackBox.decode(staticEncoded);
    
    console.log('✓ Static encode/decode test passed');
    console.log('Original:', testData);
    console.log('Encoded:', staticEncoded.substring(0, 50) + '...');
    console.log('Decoded:', staticDecoded);

    // Utility functions demonstration
    console.log('\n--- Utility Functions ---');
    console.log('Current time (ISO):', Utils.getCurrentTimeISO());
    console.log('Current time (ms):', Utils.getCurrentTimeMs());
    console.log('Random string (16):', Utils.randomString(16));
    
    const testString = 'Hello, World!';
    const base64 = Utils.base64Encode(testString);
    console.log('Base64 encode:', base64);
    console.log('Base64 decode:', Utils.base64Decode(base64));
    
    // Crypto functions
    console.log('\n--- Crypto Functions ---');
    const hash = Crypto.sha512Hex('test message');
    console.log('SHA512 hash:', hash.substring(0, 32) + '...');
    
    const encrypted = Crypto.xorEncrypt('secret', 'key');
    const decrypted = Crypto.xorDecrypt(encrypted, 'key');
    console.log('XOR encrypt/decrypt test:', decrypted === 'secret' ? 'PASSED' : 'FAILED');

    console.log('\n✓ Basic example completed successfully!');

  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

// Run the example
basicExample().catch(console.error);

File: gfless-cpp/examples/basic_example.cpp
#include "gfless/gfless.h"
#include <iostream>

int main() {
    std::cout << "Gfless C++ Library Example\n";
    std::cout << "Version: " << gfless::VERSION << "\n\n";
    
    try {
        // Create an identity (this will create identity.json if it doesn't exist)
        auto identity = gfless::createIdentity("identity.json");
        
        // Update the identity with current timestamps
        identity->update();
        
        // Create a basic blackbox
        gfless::json request = gfless::json::object();
        request["app"] = "example";
        request["version"] = "1.0";
        
        auto blackBox = gfless::createBlackBox(identity, request);
        std::string encoded = blackBox->encoded();
        
        std::cout << "Generated BlackBox:\n";
        std::cout << encoded << "\n\n";
        
        // Demonstrate decode functionality
        std::string testData = R"(["test", 123, true])";
        std::string encodedTest = gfless::BlackBox::encode(testData);
        std::string decodedTest = gfless::BlackBox::decode(encodedTest);
        
        std::cout << "Encode/Decode Test:\n";
        std::cout << "Original: " << testData << "\n";
        std::cout << "Encoded:  " << encodedTest << "\n";
        std::cout << "Decoded:  " << decodedTest << "\n\n";
        
        // Display some utility functions
        std::cout << "Utility Functions:\n";
        std::cout << "Current time (ISO): " << gfless::Utils::getCurrentTimeISO() << "\n";
        std::cout << "Current time (ms):  " << gfless::Utils::getCurrentTimeMs() << "\n";
        std::cout << "Random string:      " << gfless::Utils::randomString(16) << "\n";
        
        std::cout << "\nExample completed successfully!\n";
        
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }
    
    return 0;
}